### PR TITLE
Initial sample Play app - with github actions integration for build/test

### PR DESCRIPTION
**Pull request by ndmckinley**
This includes the basic docker container which can run this application.  No decisions set in stone about that container - this is just the simplest thing that could work.  Plenty of things to consider optimizing later.

We chose the standard Play Getting Started app for the baseline.  We think that'll make it easier for folks to make some progress, since it already does _something_.  :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+logs/
+target/
+RUNNING_PID
```
Comment (bion) [inline .gitignore:3]: any chance we can put this in a local `tmp` dir instead of the root?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# AlpineLinux with a glibc-2.23, Oracle Java 8, sbt and git
+FROM anapsix/alpine-java:8_jdk
```
Comment (bion) [inline Dockerfile:2]: thoughts on using Java 11 LTS instead of 8?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Install sbt
+RUN apk add --no-cache --update bash wget && mkdir -p "$SBT_HOME" && \
+    wget -qO - --no-check-certificate "https://dl.bintray.com/sbt/native-packages/sbt/$SBT_VERSION/sbt-$SBT_VERSION.tgz" |  tar xz -C $INSTALL_DIR && \
+    echo -ne "- with sbt $SBT_VERSION\n" >> /root/.built
+
+# Install git
+RUN apk add --no-cache git openssh
+
+# Install node.js
+RUN apk add nodejs
```
Comment (bion) [inline Dockerfile:21]: should we specify a version?

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (bion) [inline LICENSE:1]: this should already exist

File: README.md [hunk 5]
```diff
@@ -1 +1,8 @@
-# universal-application-tool
\ No newline at end of file
+# universal-application-tool
+
+
+To build the container that runs the app, type `docker build -t uat .`
+
+To run the application, run `docker run -it -p 9000:9000 -v ${PWD}/universal-application-tool-0.0.1:/code uat sbt ~run`
```
Comment (bion) [inline README.md:6]: let's put this in a script at `bin/run-dev` for convenience

File: universal-application-tool-0.0.1/app/controllers/AsyncController.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import scala.concurrent.duration.Duration;
+import scala.concurrent.ExecutionContextExecutor;
+
+/**
+ * This controller contains an action that demonstrates how to write
+ * simple asynchronous code in a controller. It uses a timer to
+ * asynchronously delay sending a response for 1 second.
+ */
+@Singleton
+public class AsyncController extends Controller {
```
Comment (bion) [inline universal-application-tool-0.0.1/app/controllers/AsyncController.java:24]: It'll be worth discussing as a team whether we want to implement all I/O async. It's certainly preferable from a throughput and scalability perspective, but I doubt the app will ever have enough traffic for that to be worth the tradeoff in development velocity.

**Review by bion — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (ndmckinley) [reply .gitignore:3]: Do you mean the whole gitignore?  gitignore is shared / global.
https://git-scm.com/docs/gitignore

File: Dockerfile (hunk 2 above)
Comment (ndmckinley) [reply Dockerfile:2]: No opinion - is it better in some way?

It's a little hard to get a docker image with a working oracle jdk in it, and someone already did 99% of the work for us with java 8 but not so with java 11.

File: Dockerfile (hunk 3 above)
Comment (ndmckinley) [reply Dockerfile]: Done.

File: LICENSE (hunk 4 above)
Comment (ndmckinley) [reply LICENSE]: It totally does - this was a typo, re-merging your commit into our history.  I can cherrypick it out but it'll require a force-push.

File: README.md (hunk 5 above)
Comment (ndmckinley) [reply README.md:6]: Done.

File: Dockerfile (hunk 2 above)
Comment (bion) [reply Dockerfile:2]: Let's leave it at 8 for now. I brought it up in the interest of pushing out the day when the language version will be out of date but getting up and running is higher priority right now.

File: .gitignore (hunk 1 above)
Comment (bion) [reply .gitignore:3]: I wasn't thinking straight with this comment, feel free to ignore

File: Dockerfile (hunk 3 above)
Comment (ndmckinley) [reply Dockerfile]: reviewed alpine recommendations, they suggest not doing this unless you've proven it's needed - the packages are regularly dropped from the repo and dockerfile builds will fail.

https://gitlab.alpinelinux.org/alpine/abuild/-/issues/9996#note_87135

File: Dockerfile (hunk 2 above)
Comment (ndmckinley) [reply Dockerfile:2]: Did a little more reading - I think that openjdk works fine with sbt, so I'm experimenting.  openjdk is so much easier!  :)

Comment (ndmckinley) [reply Dockerfile:2]: Great, all the tests still pass with it.  :)

File: Dockerfile (hunk 3 above)
Comment (cityofseattle-googlefellowship) [reply Dockerfile:22]: thanks for looking more closely at this!

**Pull request merged**